typescript/sdk/src/token/deploy.ts

```diff
@@ -330,54 +319,31 @@ export class HypERC721Deployer extends GasRouterDeployer<
     chain: ChainName,
     config: HypERC721CollateralConfig,
   ): Promise<HypERC721Collateral> {
-    let router: HypERC721Collateral;
-    if (isUriConfig(config)) {
-      router = await this.deployContractFromFactory(
-        chain,
-        new HypERC721URICollateral__factory(),
-        'HypERC721URICollateral',
-        [config.token, config.mailbox],
-      );
-    } else {
-      router = await this.deployContractFromFactory(
-        chain,
-        new HypERC721Collateral__factory(),
-        'HypERC721Collateral',
-        [config.token, config.mailbox],
-      );
-    }
-    return router;
+    return (await this.deployContract(
+      chain,
+      getTokenType(config.type, TokenStandard.ERC721),
+      [config.token, config.mailbox],
+    )) as HypERC721Collateral;
   }
 
   protected async deploySynthetic(
     chain: ChainName,
     config: HypERC721Config,
   ): Promise<HypERC721> {
-    let router: HypERC721;
-    if (isUriConfig(config)) {
-      router = await this.deployContractFromFactory(
-        chain,
-        new HypERC721URIStorage__factory(),
-        'HypERC721URIStorage',
-        [config.mailbox],
-      );
-    } else {
-      router = await this.deployContractFromFactory(
-        chain,
-        new HypERC721__factory(),
-        'HypERC721',
-        [config.mailbox],
-      );
-    }
-    await this.multiProvider.handleTx(
+    return (await this.deployContract(
       chain,
-      router.initialize(config.totalSupply, config.name, config.symbol),
-    );
-    return router;
+      getTokenType(config.type, TokenStandard.ERC721),
+      [config.mailbox],
+    )) as HypERC721;
   }
 
   router(contracts: HyperlaneContracts<HypERC721Factories>) {
-    return contracts.router;
+    for (const key of Object.keys(hypERC721factories)) {
+      if (contracts[key as keyof typeof contracts]) {
+        return contracts[key as keyof typeof contracts];
```

Comment:
A few things that may help here:
- Set an explicit type on `hypERC721factories` where it's defined.
- Add `as const` after `hypERC721factories` definition
- Add this method to the utils object.ts file which maintains the key type. I thought we had this already but apparently not:
```ts
// Useful for maintaining type safety when using Object.keys
export function objKeys<T extends string | number>(obj: Record<T, any>): T[] {
  return Object.keys(obj) as T[];
}
```

Reply:
there is a global `PropertyKey` type or something